javascript/ql/src/semmle/javascript/security/dataflow/CommandInjectionCustomizations.qll

```diff
@@ -28,7 +28,10 @@ module CommandInjection {
 
   /** A source of remote user input, considered as a flow source for command injection. */
   class RemoteFlowSourceAsSource extends Source {
-    RemoteFlowSourceAsSource() { this instanceof RemoteFlowSource }
+    RemoteFlowSourceAsSource() {
+      this instanceof RemoteFlowSource and
+      not this instanceof ClientSideRemoteFlowSource
+    }
```

Comment:
```suggestion
    RemoteFlowSourceAsSource() { this.(RemoteFlowSource).isServerSource() }
```

How about adding `final` helper predicates to RemoteFlowSource:  `isServerSource()` and `isClientSource()`?   
(The implementation of `isServerSource` would just be `not isClientSource()`).

Reply:
As mentioned in the PR description, my main gripe with that is that we can't really guarantee whether something is a server-side taint source.

Reply:
Yeah, and I supose `isASourceThatMightAppearInAServer` is too much of a mouthful.  

LGTM. But you got a conflict.